### PR TITLE
Add enum for modes of locking buttons

### PR DESCRIPTION
**Pull request by ia**
* **Please check if the PR fulfills these requirements**
- [x] The changes have been tested locally
- [x] There are no breaking changes

* **What kind of change does this PR introduce?**
Add `enum` for modes of locking buttons in _Settings_.

* **What is the current behavior?**
Constants are used for `SettingsOptions::LockingMode`.

* **What is the new behavior (if this is a feature change)?**
Use _self-documented_ `enum` values for `SettingsOptions::LockingMode`.

* **Other information**:
With this tiny refactoring now all the constants for settings should be _ironed out_ ;)

### CODE REVIEW

File: source/Core/Inc/Settings.h [hunk 1]
```diff
@@ -110,6 +110,12 @@ typedef enum {
   NO_DYNAMIC = 0, // PPS + EPR disabled, fixed PDO only
 } usbpdMode_t;
 
+typedef enum {
+  DISABLED = 0, // Locking buttons is disabled
+  BOOST    = 1, // Locking buttons for Boost mode only
+  FULL     = 2, // Locking buttons fully
```
Comment (Ralim) [inline source/Core/Inc/Settings.h:116]:
```suggestion
  FULL     = 2, // Locking all buttons
```

Comment (ia) [reply source/Core/Inc/Settings.h:116]: Are you sure that this will be better? I just tried to keep the code as coherent as possible in accordance with the other parts including [description itself](https://github.com/Ralim/IronOS/blob/84e53e11e0835589d88fd4c231cc68883fe74fff/Translations/translation_EN.json#L158) and [documentation](https://github.com/Ralim/IronOS/blob/84e53e11e0835589d88fd4c231cc68883fe74fff/Documentation/Settings.md?plain=1#L249) to avoid any unnecessary ambiguity and not to puzzle anyone without extra need in the future.

We could change all the other descriptions after the suggestion though, but it would lead to the need of updating all translation files as well, which is - in my very humble opinion - total overkill.

TL;DR: this option-related value is notated as `full` in _the description_ and in _the documentation_ so why it should be `all` in _the code_ exactly? Just curious.

Comment (Ralim) [reply source/Core/Inc/Settings.h:116]: It can be full, mostly just wanted to not have "Locking buttons fully" as it feels a very clunky description, and we are not length limited here like we are in the description.

Comment (ia) [reply source/Core/Inc/Settings.h:116]: Aha, ok. I think I got it... Let me prepare a slightly updated solution then.

File: source/Core/Inc/Settings.h [hunk 2]
```diff
@@ -113,7 +113,7 @@ typedef enum {
 typedef enum {
   DISABLED = 0, // Locking buttons is disabled
   BOOST    = 1, // Locking buttons for Boost mode only
-  FULL     = 2, // Locking buttons fully
+  FULL     = 2, // Locking buttons for Boost mode AND for Soldering mode
```
Comment (ia) [inline source/Core/Inc/Settings.h:116]: How about this? :point_right: :point_left:

Comment (Ralim) [reply source/Core/Inc/Settings.h:116]: 👍🏼

**Review by Ralim — APPROVED** [review]

**Comment by neon12345** [comment]
Is there a missing delay between lock and unlock? If I hold both buttons, it very quickly toggles between locked and unlocked.

**Comment by Ralim** [comment]
@neon12345 Best to open a new issue for this as this PR has already been merged and is not directly related.

**Comment by neon12345** [comment]
@Ralim see #1956
